Mathlib/Order/OrderIsoNat.lean

```diff
@@ -292,3 +292,14 @@ theorem exists_covBy_seq_of_wellFoundedLT_wellFoundedGT (α) [Preorder α]
     by_contra!
     exact (RelEmbedding.natGT a fun n ↦ (cov n (this n)).1).not_wellFounded_of_decreasing_seq wfg.wf
   exact ⟨_, wellFounded_lt.min_mem _ H, fun i h ↦ cov _ fun h' ↦ wellFounded_lt.not_lt_min _ H h' h⟩
+
+theorem exists_covBy_seq_of_wellFoundedLT_wellFoundedGT_of_le {α : Type*} [PartialOrder α]
+    [wfl : WellFoundedLT α] [wfg : WellFoundedGT α] {x y : α} (h : x ≤ y) :
```

Comment:
Are there interesting orders that show up which are well founded in both directions? It seems to me that they're all disjoint unions of finite orders, and so `h : x <= y` means you can restrict to one of these finite suborders anyway

Reply:
Perhaps the ideal lattice of any artinian ring (e.g. $$\mathbb{Q}[x, y]/(x^2, xy, y^2)$$).

Reply:
Ah! Right, they don't have to be a disjoint union; they're just posets of finite height.